apps/sync/js/adapters/history.js

```diff
@@ -0,0 +1,98 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+/* global IAC,
+  SyncEngine
+*/
+
+var HistoryHelper = {
+  set syncedCollectionModifiedTime(mtime) {
+    window.asyncStorage.setItem('LastSyncedStatus::Collection::mtime', mtime);
+  },
+  get syncedCollectionModifiedTime() {
+    return new Promise((resolve, reject) => {
+      window.asyncStorage.
+      getItem('LastSyncedStatus::Collection::mtime', (mtime) => {
+        resolve(mtime);
+      });
+    });
+  },
+
+  addPlaces(places) {
+    return IAC.request('sync-history', {
+      method: 'addPlaces',
+      args: [places]
+    });
+  },
+
+  addPlace(place) {
+    return IAC.request('sync-history', {
+      method: 'addPlace',
+      args: [place]
+    });
+  }
+};
+
+SyncEngine.DataAdapterClasses.history = {
+  _fullSync(kintoCollection, lastModifiedTime) {
+    function updateHistoryCollection(list) {
+      var historyRecords = list.data;
+      var places = [];
+      var i;
+      for (i = 0; i < historyRecords.length; i++) {
+        if (historyRecords[i].last_modified <= lastModifiedTime) {
```

Comment:
Also, now all records have `last_modified` because we are taking everything from the server (readonly). But it would be good to shield it just in case we have records with a missing `last_modified` property.

From kinto.js doc: `Records with last_modified attribute were sync'ed on a server.`